src/Container/LCParticleContainer.cpp

```diff
@@ -222,15 +284,21 @@ std::vector<Particle *> LCParticleContainer::getBoundaryParticles() {
   return result;
 }
 
+
 void LCParticleContainer::handleBoundaryAction() {
   std::vector<Particle*> boundaryparticles = getBoundaryParticles();
+  std::array<double, 12> bounds;
     for(auto p: boundaryparticles) {
-      std::array<double, 6> bounds = getInfluencingBoundarysWithDistance(p);
+      bounds = getInfluencingBoundarysWithDistance(p);
         for(int i = 0; i < 6; i++) {
-          if(boundary_types[i] == 1){} //outflow
+          if(boundary_types[i] == 1) {
+            if(bounds.at(i) < 0) {
+              p->park();
+            }
+          } //outflow
           else if(boundary_types[i] == 2){
             // generate halo Particle
-            if(bounds.at(i) != -1.0 && abs(bounds.at(i)) <= r_cutoff){ //&& abs(bounds.at(i)) <= r_cutoff
+            if(bounds.at(i) != -1.0 && abs(bounds.at(i)) <= pow(2, 1/6)){ //&& abs(bounds.at(i)) <= r_cutoff
               if( i == 0 ){ //Boundary to YZ Plane
                 std::array<double, 3> x_arg = {-bounds.at(i), p->getX().at(1), p->getX().at(2)};
```

Comment:
potential error: position of halo particle in case of reflection is not at boundary but instead it is mirrored across boundary, i.e. `x_arg[0] = bounds.at(i) + (bounds.at(i) - p->getX()[0])` or something like this

Reply:
Ok, so here you are on YZ plane. So, y and z coordinate of halo particles will be same the actual particle, which I see is correct here. However, as you said, for direction, you need to mirror it. so, x coordinate here should be `- p->getX()[0]` right? even if it is mirrored about `x=0`. and not `- bounds[0]` which is exactly at the boundary. Or did I misunderstand it?

The formula I was writing to write would be a more general case where domain doesn't necessarily start at `x=0`.

Reply:
i think my naming for the Bounds variable is a bit confusing. it contains the distances to the boundary, the particle is close to. this value is only necessary for the Borders, that are not touching the center at 0,0,0. So I do set the X Value for our Halo Particle to -p->getX() but with a little extra effort. I think thats not a big decrease in performance because the calculation of this -X is implemented in the getInfluencingBoundarysWithDistance Methode and not that big.

Reply:
Ah, I see. It is fine then. Thanks for the clarification again :)